packages/tools/viewer-alpha/src/asyncLock.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Provides a simple way of creating the rough equivalent of an async critical section.
+ *
+ * @example
+ * ```typescript
+ * const myLock = new AsyncLock();
+ *
+ * private async MyFuncAsync(): Promise<void> {
+ *   await myLock.lockAsync(async () => {
+ *     await operation1Async();
+ *     await operation2Async();
+ *   });
+ * }
+ * ```
+ */
+export class AsyncLock {

```

Comment:
This is a useful generic utility for async code, and I expect to add a bit more to it (`AbortSignal` and probably a `lockManyAsync` for "shutdown" scenarios). I think it would probably be better to just put this in Core under Misc, near `Deferred` and other async utilities. Any objections?

Reply:
makes sense.